TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyDecodedHttpRequestHandler.java

```diff
@@ -28,59 +27,79 @@ public class NettyDecodedHttpRequestHandler extends ChannelInboundHandlerAdapter
             chunkSizes.add(new ArrayList<>(32));
             var request = (HttpRequest) msg;
             var httpJsonMessage = parseHeadersIntoMessage(request);
             var pipelineOrchestrator = new RequestPipelineOrchestrator(chunkSizes, packetReceiver);
             var pipeline = ctx.pipeline();
             try {
-                transformer.transformJson(httpJsonMessage);
-                if (headerFieldIsIdentical("content-encoding", request, httpJsonMessage) &&
-                        headerFieldIsIdentical("transfer-encoding", request, httpJsonMessage)) {
-                    statusWatcher.accept(HTTP_CONSUMPTION_STATUS.DONE);
-                    // TODO - there might be a memory leak here.
-                    // I'm not sure if I should be releasing the HttpRequest `msg`.
-                    log.info("TODO - there might be a memory leak here.  " +
-                            "I'm not sure if I should be releasing the HttpRequest `msg`.");
-                    pipelineOrchestrator.addBaselineHandlers(pipeline);
-                    ctx.fireChannelRead(httpJsonMessage);
-                    pipelineOrchestrator.removeThisAndPreviousHandlers(pipeline, this);
-                    return;
-                } else {
-                    pipelineOrchestrator.addContentRepackingHandlers(pipeline);
-                }
+                handlePayloadNeutralTransformation(ctx, request, httpJsonMessage, pipelineOrchestrator);
             } catch (PayloadNotLoadedException pnle) {
+                log.debug("The transforms for this message require payload manipulation, " +
+                        "all content handlers are being loaded.");
                 // make a fresh message and its headers
                 httpJsonMessage = parseHeadersIntoMessage(request);
                 pipelineOrchestrator.addJsonParsingHandlers(pipeline, transformer);
+                ctx.fireChannelRead(httpJsonMessage);
             }
-            // send both!  We'll allow some built-in netty http handlers to do their thing & then
-            // reunify any additions with our headers model before serializing
-            ctx.fireChannelRead(request);
-            ctx.fireChannelRead(httpJsonMessage);
         } else if (msg instanceof HttpContent) {
-            if (msg instanceof LastHttpContent) {
-                statusWatcher.accept(HTTP_CONSUMPTION_STATUS.DONE);
-            }
             ctx.fireChannelRead(msg);
         }
     }
 
+    private void handlePayloadNeutralTransformation(ChannelHandlerContext ctx,
+                                                    HttpRequest request,
+                                                    HttpJsonMessageWithFaultablePayload httpJsonMessage,
+                                                    RequestPipelineOrchestrator pipelineOrchestrator)
+            throws PayloadNotLoadedException
+    {
+        var pipeline = ctx.pipeline();
+        transformer.transformJson(httpJsonMessage);
```

Comment:
What is this command actually doing as we don't directly check a response?

Reply:
We do the transformation but may later completely disregard it, is this correct?

Reply:
This was updated in PR-182 to reset the httpJsonMessage.
In _some_ cases we'll disregard it.  Line 82 will send it along as is, before a reset though.  That worked before because the transformations today won't replace the top-level object.  The signature indicates that that could happen, so you (& kartg) are right that it should be updated (and it is in a future commit)